tests/unit/charms/tls_certificates_interface/v4/test_tls_certificates_v4.py

```diff
@@ -369,6 +369,24 @@ def test_given_certificate_signin_request_when_from_csr_then_attributes_are_corr
     assert attributes.locality_name == "Montreal"
 
 
+def test_generated_ca_does_not_have_an_empty_sans_extension():
+    ca_private_key = generate_private_key()
+    ca_certificate = generate_ca(
+        private_key=ca_private_key,
+        validity=timedelta(days=365),
+        common_name="certifier.example.com",
+    )
+    ca = x509.load_pem_x509_certificate(ca_certificate.raw.encode())
+    alt_name_ext = [e for e in ca.extensions if e.oid._name == "subjectAltName"]
+
+    # It's OK for the extension to be missing
+    assert len(alt_name_ext) in (0, 1)
+
+    if alt_name_ext:
+        # If the subjectAltName extension is present, it cannot be empty
+        assert alt_name_ext[0].value._general_names[:]
```

Comment:
I'm not sure I understand this test. Shouldn't we be testing the expected behaviour, as opposed to just asserting the possible valid configurations?

In other words, should we not test that the alt name extension is missing since we don't provide any SANS to `generate_ca()`?

Reply:
Ok, that makes sense.